src/code/z_bgcheck.c

```diff
@@ -2952,139 +2945,145 @@ void DynaPoly_ExpandSRT(PlayState* play, DynaCollisionContext* dyna, s32 bgId, s
+                newPoly->normal.x = COLPOLY_SNORMAL(newNormal.x);
+                newPoly->normal.y = COLPOLY_SNORMAL(newNormal.y);
+                newPoly->normal.z = COLPOLY_SNORMAL(newNormal.z);
+            }
 
-                newPoly->dist = func_80086D24(-DOTXYZ(newNormal, vtxA));
-                if (newNormal.y > 0.5f) {
-                    s16 polyId = *polyStartIndex + i;
+            newPoly->dist = func_80086D24(-DOTXYZ(newNormal, vtxA));
+            if (newNormal.y > 0.5f) {
+                s16 polyId = *polyStartIndex + i;
 
-                    DynaSSNodeList_SetSSListHead(&dyna->polyNodes, &dyna->bgActors[bgId].dynaLookup.floor, &polyId);
-                } else if (newNormal.y < -0.8f) {
-                    s16 polyId = *polyStartIndex + i;
+                DynaSSNodeList_SetSSListHead(&dyna->polyNodes, &dyna->bgActors[bgId].dynaLookup.floor, &polyId);
+            } else if (newNormal.y < -0.8f) {
+                s16 polyId = *polyStartIndex + i;
 
-                    DynaSSNodeList_SetSSListHead(&dyna->polyNodes, &dyna->bgActors[bgId].dynaLookup.ceiling, &polyId);
-                } else {
-                    s16 polyId = *polyStartIndex + i;
+                DynaSSNodeList_SetSSListHead(&dyna->polyNodes, &dyna->bgActors[bgId].dynaLookup.ceiling, &polyId);
+            } else {
+                s16 polyId = *polyStartIndex + i;
 
-                    DynaSSNodeList_SetSSListHead(&dyna->polyNodes, &dyna->bgActors[bgId].dynaLookup.wall, &polyId);
-                }
+                DynaSSNodeList_SetSSListHead(&dyna->polyNodes, &dyna->bgActors[bgId].dynaLookup.wall, &polyId);
             }
         }
+    }
 
-        if (pbgdata->numWaterBoxes > 0) {
-            for (wi = 0; wi < pbgdata->numWaterBoxes; wi++) {
-                Math_Vec3s_ToVec3f(&spB8, &pbgdata->waterBoxes[wi].minPos);
-                Math_Vec3f_Copy(&spA0, &spB8);
-                spA0.x += pbgdata->waterBoxes[wi].xLength;
-                Math_Vec3f_Copy(&sp88, &spB8);
-                sp88.z += pbgdata->waterBoxes[wi].zLength;
-                SkinMatrix_Vec3fMtxFMultXYZ(&D_801EDA40, &spB8, &spAC);
-                SkinMatrix_Vec3fMtxFMultXYZ(&D_801EDA40, &spA0, &sp94);
-                SkinMatrix_Vec3fMtxFMultXYZ(&D_801EDA40, &sp88, &sp7C);
-                waterBox = &dyna->waterBoxList.boxes[*waterBoxStartIndex + wi];
-                BgCheck_CalcWaterboxDimensions(&spAC, &sp94, &sp7C, &waterBox->minPos, &waterBox->xLength,
-                                               &waterBox->zLength);
-                waterBox->properties = pbgdata->waterBoxes[wi].properties;
-            }
+    if (pbgdata->numWaterBoxes > 0) {
+        for (wi = 0; wi < pbgdata->numWaterBoxes; wi++) {
+            WaterBox* waterBox;
+            Vec3f spB8; // waterbox ?
+            Vec3f spAC;
+            Vec3f spA0; // waterbox ?
+            Vec3f sp94;
+            Vec3f sp88; // waterbox ?
+            Vec3f sp7C;
```

Comment:
I imagine it shouldn't be too hard to give names to these variables now.

D_801EDA40 is the model to world space matrix for the actor being processed, so the SkinMatrix_Vec3fMtxFMultXYZ calls are just transposing some bounding points of the water box. 

This means that...
* spB8, spAC is waterbox's minPos in model, world space respectively
* spA0, sp94 is minPos + (xLength, 0, 0) in model, world space
* sp88, sp7C is minPos + (0, 0, zLength) in model, world space

After the transformation, points spAC, sp94, sp7C are used to compute the new minPos and x,z dimensions of the newly created waterbox.

Reply:
Went ahead and named them

Reply:
looks good enough to me